addons/account/models/account_payment.py

```diff
@@ -672,6 +676,99 @@ def _compute_payment_receipt_title(self):
+
+        if not payments:
+            return {}
+
+        # Update tables involved in the query
+        self.env['account.move.line'].flush_model(('move_id', 'payment_id', 'balance', 'account_id', 'company_id', 'date', 'partner_id'))
+        self.env['account.payment'].flush_model(('move_id', 'payment_type'))
+
+        move_table_and_alias = SQL("""
+            (SELECT account_move_line.*, account_payment.payment_type
+               FROM account_move_line
+               JOIN account_payment
+                 ON account_payment.move_id = account_move_line.move_id) AS move_line
+        """)
+        suspense_account_id = payments.company_id.account_journal_suspense_account_id.id
+
+        if not self[0].id:  # if record is under creation/edition in UI, safely inject values in the query
+            # Necessary since new record aren't searchable in the DB and record in edition aren't up to date yet
+            place_holders = {
+                'move_id': self._origin.move_id.id or 0,
+                'payment_id': self._origin.id or 0,
+                'payment_type': self.payment_type,
+                'balance': self.amount if self.payment_type == 'outbound' else -self.amount,
+                'account_id': self.destination_account_id.id,
+                'company_id': self.company_id.id or None,
+                'date': self.date or None,
+                'partner_id': self.partner_id.id,
+            }
+            move_table_and_alias = SQL("""
+                (VALUES (%(move_id)s, %(payment_id)s, %(company_id)s, %(date)s, %(partner_id)s, %(balance)s, %(payment_type)s, %(account_id)s))
+                AS move_line(move_id, payment_id, company_id, date, partner_id, balance, payment_type, account_id)
+            """, **place_holders)
+
+        query = SQL(
+            """
+            SELECT
+                   move_line.payment_id,
+                   array_agg(dup_move_line.move_id) AS duplicate_move_ids
+              FROM %(move_table_and_alias)s
+              JOIN account_move_line AS dup_move_line
+                ON move_line.move_id != dup_move_line.move_id
+               AND move_line.partner_id = dup_move_line.partner_id
+               AND move_line.company_id = dup_move_line.company_id
+               AND move_line.date = dup_move_line.date
+               AND dup_move_line.parent_state IN %(matching_states)s
+               AND move_line.balance = dup_move_line.balance
+               AND (
+                   move_line.account_id = dup_move_line.account_id
+                   OR dup_move_line.account_id = %(suspense_account_id)s
+               )
+             WHERE move_line.payment_id IN %(payments)s
+               -- Don't include moves rows where any of its move lines is reconciled.
+               AND dup_move_line.move_id NOT IN (
+                   SELECT move_id FROM account_move_line WHERE account_move_line.reconciled
+               )
```

Comment:
Can't we use `is_reconciled`?
Not sure what the performance of this is

Reply:
I tried with amount_residual but the same move line that has reconciled = true has amount residual = 0, so I had the same problem where the move line that matches the account we're interested in doesn't indicate that the move line is reconciled. So I went back to a previous approach, filtering out moves that have reconciled entries.